Initium-Odp/src/com/universeprojects/miniup/server/commands/CommandAttack.java

```diff
@@ -0,0 +1,200 @@
+package com.universeprojects.miniup.server.commands;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.universeprojects.cacheddatastore.CachedDatastoreService;
+import com.universeprojects.cacheddatastore.CachedEntity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.universeprojects.miniup.server.GameUtils;
+import com.universeprojects.miniup.server.ODPDBAccess;
+import com.universeprojects.miniup.server.commands.framework.Command;
+import com.universeprojects.miniup.server.commands.framework.UserErrorMessage;
+
+
+/**
+ * Allows the player to attack a mob at the current location
+ * 
+ * Usage notes:
+ * Checks if caller can attack and initiates combat if so
+ * 
+ * Parameters:
+ * 		charId - characterId of monster to attack 
+ * 
+ * Support methods:
+ * 		canAttack - returns boolean whether attack is allowed
+ * 
+ * @author NJ
+ *
+ */
+public class CommandAttack extends Command {
+
+	public CommandAttack(HttpServletRequest request, HttpServletResponse response) 
+	{
+		super(request, response);
+	}
+
+	public static boolean canAttack(CachedDatastoreService ds, CachedEntity location, CachedEntity character, CachedEntity monster)
+	{
+		try {
+			canAttackHelper(ds, location, character, monster);
+			return true;
+		} catch (Exception e) {
+			return false;
+		}
+	}
+	
+	private static void canAttackHelper(CachedDatastoreService ds, CachedEntity location, CachedEntity character, CachedEntity monster) throws UserErrorMessage
+	{
+		if (ds==null || location==null || character==null || monster==null)
```

Comment:
This is only here for the canAttack call really, and there any error is meaningless other than returning false. The 1 specific case where monster could be null from the Attack command is checked before calling the function.